### ISSUE TITLE
Clean up main.py

### ISSUE THREAD

**Issue by Davidy22** [Davidy22/SkunkBooth#61]
main.py has become a bit of a dumping ground, clean it up a little bit by moving what should be moved out into other files.

**Comment by Arnab1181412** [comment]
Hey I want to fix this issue im new to github contributor and have good knowledge of python and can you give some context to it

**Comment by rashidkalwar** [comment]
yeah I can see it is a bit mess in there. I can help solving it, assign it to me. I will make a PR as soon as I can.

**Comment by Davidy22** [comment]
Oh whoops, github scrolled me to the bottom so I only saw rashid before I assigned. Well, there's another similar ticket so I can assign that one to arnab

**Comment by rashidkalwar** [comment]
hey @Davidy22
I made some changes and then made a pull request but I saw in the post commit their were some spacing and indentation issues, I checked and fixed all, still facing errors, please have a look where am I messing wrong.

**Comment by rashidkalwar** [comment]
sorry for all that mess with commits

**Comment by Davidy22** [comment]
You can run `pre-commit install` on the top level directory to install hooks that'll check your code on your machine and tell you what lines need checking.

**Comment by rashidkalwar** [comment]
@Davidy22 
just closed the last PR, and made a new one this time looks great, please have a look and let me know.

**Comment by rashidkalwar** [comment]
@Davidy22 I am having a hard time understanding it and honestly I don't know what can mess wrong, since you have the access to make changes to my Pull Request, please modifying it so it doesn't crash anything.

**Comment by Davidy22** [comment]
I can take a look later after I get some stuff I wanted to clear done, but this should just be a matter of passing a reference to the buffer into the function.

**Comment by Davidy22** [comment]
Still busy with my stuff but it seems like rashid's having trouble with this, I'll assign both of you to this since I think github hasn't pinged arnab for the other ticket I mentioned him in, you can try fix up rashid's attempt or start from scratch.
